lib/inspector.js

```diff
@@ -160,6 +160,49 @@ class Session extends EventEmitter {
   }
 }
 
+class RawSession extends EventEmitter {
+  constructor() {
+    super();
+    this[connectionSymbol] = null;
+  }
+
+  connect() {
+    if (this[connectionSymbol])
+      throw new ERR_INSPECTOR_ALREADY_CONNECTED('The inspector session');
+    this[connectionSymbol] =
+      new Connection((message) => this[onMessageSymbol](message));
+  }
+
+  connectToMainThread() {
+    if (isMainThread)
+      throw new ERR_INSPECTOR_NOT_WORKER();
+    if (this[connectionSymbol])
+      throw new ERR_INSPECTOR_ALREADY_CONNECTED('The inspector session');
+    this[connectionSymbol] =
+      new MainThreadConnection(
+        (message) => queueMicrotask(() => this[onMessageSymbol](message)));
+  }
+
+  [onMessageSymbol](message) {
+    this.emit('message', message);
+  }
+
+  post(message) {
+    validateString(message);
+    if (!this[connectionSymbol]) {
+      throw new ERR_INSPECTOR_NOT_CONNECTED();
+    }
+    this[connectionSymbol].dispatch(message);
+  }
+
+  disconnect() {
+    if (!this[connectionSymbol])
+      return;
+    this[connectionSymbol].disconnect();
+    this[connectionSymbol] = null;
+  }
+}
```

Comment:
If this PR lands (and it might be good to clarify that first), it would probably make sense to implement `Session` on top of `RawSession` instead of duplicating all logic that is in `RawSession`.

Reply:
Yes, i will re-implement the `Session` based on `RawSession`.